### PR TITLE
Fix - Megamenu not closing

### LINKED ISSUE #1538: 🐛 New Mega Menu - Make dropdown close after a click
cc @hveraus @wicksipedia @adamcogan 

The new menu looks awesome. However, we need to make the dropdown automatically close after a click... 
Keeping it open is unexpected (bug) + makes it hard to see the selected page has actually open 

**To Reproduce**
1. Click on `Services` from the navbar 
2. Menu - Click on `Application Development` 
3. Menu - Now click on `Website Development` - dropdown will stay open here 

**Expected behavior**
It should close its menu when navigating to a new page. 

 
- [x] 1. Please fix

https://github.com/SSWConsulting/SSW.Website/assets/12601228/12780ab8-82b1-4322-8fd0-5a16633281f6

### PR DESCRIPTION
**Pull request by amankumarrr**
Affected routes: 
`/*`

- Fixed #1538 

- [x] Include done video or screenshots 
7A0A2901-14A7-45EC-A51A-4D53AFA95F72

**GIF: Showing Megamenu being closed on navigation**

### CODE REVIEW

File: components/megamenu/MegaMenuLayout/mega-menu-layout.tsx [hunk 1]
```diff
@@ -53,11 +53,13 @@ const MegaMenuLayout: React.FC<MegaMenuWrapperProps> = ({
             />
             <Search />
             <Divider />
+            asdfasf
             <button
               type="button"
               className="-m-2.5 inline-flex items-center justify-center rounded-md pl-6 pr-2 text-gray-700"
               onClick={() => setMobileMenuOpen(true)}
             >
+              asdfasdfdsafasdf
```
Comment (bradystroud) [inline components/megamenu/MegaMenuLayout/mega-menu-layout.tsx]: 🤔

**Review by bradystroud — CHANGES_REQUESTED** [review]

File: components/megamenu/MegaMenuLayout/mega-menu-layout.tsx (hunk 1 above)
Comment (amankumarrr) [reply components/megamenu/MegaMenuLayout/mega-menu-layout.tsx]: Just cleaned up

File: components/megamenu/DesktopMenu/desktop-menu.tsx [hunk 2]
```diff
@@ -13,6 +13,8 @@ export interface DesktopMenuProps {
   menuBarItems: NavMenuItem[];
 }
 
+export const ClosePopoverContext = createContext(null);
```
Comment (Harry-Ross) [inline components/megamenu/DesktopMenu/desktop-menu.tsx:16]: Can you move this context to the context folder?

**Review by bradystroud — APPROVED** [review]
LGTM (respond to @Harry-Ross comment first)

File: components/megamenu/DesktopMenu/desktop-menu.tsx (hunk 2 above)
Comment (amankumarrr) [reply components/megamenu/DesktopMenu/desktop-menu.tsx:16]: Given that we've planned to relocate our megamenu component into a standalone package, I refrained from incorporating it under the context folder. What are your thoughts on this, @Harry-Ross?

Comment (Harry-Ross) [reply components/megamenu/DesktopMenu/desktop-menu.tsx:16]: Fair enough - probably a good call then

**Review by Harry-Ross — APPROVED** [review]

**Pull request merged**